File: benchmarks/bench_primes.c
// bench_primes.c — Prime Counting, 30 reps — self-timed
#include <stdio.h>
#include <windows.h>

int is_prime(int n) {
    if (n < 2) return 0;
    for (int d = 2; (long long)d * d <= n; d++)
        if (n % d == 0) return 0;
    return 1;
}

int main(void) {
    LARGE_INTEGER freq, t0, t1;
    QueryPerformanceFrequency(&freq);
    QueryPerformanceCounter(&t0);
    int result = 0;
    for (int r = 0; r < 30; r++) {
        int count = 0;
        for (int n = 2; n <= 100000; n++)
            if (is_prime(n)) count++;
        result = count;
    }
    QueryPerformanceCounter(&t1);
    double ms = (double)(t1.QuadPart - t0.QuadPart) * 1000.0 / freq.QuadPart;
    printf("%d\n", result);
    fflush(stdout);
    fprintf(stderr, "COMPUTE_MS=%.3f\n", ms);
    return 0;
}

File: benchmarks/bench_primes.js
const { performance } = require('perf_hooks');
function is_prime(n) {
    if (n < 2) return false;
    for (let d = 2; d * d <= n; d++)
        if (n % d === 0) return false;
    return true;
}
const t0 = performance.now();
let result = 0;
for (let r = 0; r < 30; r++) {
    let count = 0;
    for (let n = 2; n <= 100000; n++)
        if (is_prime(n)) count++;
    result = count;
}
const ms = performance.now() - t0;
console.log(result);
process.stderr.write(`COMPUTE_MS=${ms.toFixed(3)}\n`);
